test/tools/router_check/source/router.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <fstream>
+
+#include "common/common/logger.h"
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/json/json_loader.h"
+#include "common/router/config_impl.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/precompiled/precompiled_test.h"
+#include "test/test_common/utility.h"
+#include "test/tools/router_check/json/tool_config_schemas.h"
+
+/** Class that store the configuration parameters of the router
+ * check tool extracted from a json input file
+ */
+class ToolConfig {
+public:
```

Comment:
This is equivalent to a `struct` if everything is public, so you could just make it a `struct` and drop `public`. I think it's simple enough that I would just do this, the alternative is to make the member fields private and add public getters, but this isn't Java.